src/plugins/google/types.ts

```diff
@@ -12,14 +12,13 @@ import { PermissionSpec } from "../../types/request";
 import { CliPermissionSpec } from "../../types/ssh";
 import { CommonSshPermissionSpec } from "../ssh/types";
 
-export type GcpSshPermission = {
-  spec: CommonSshPermissionSpec & {
+export type GcpSshPermission = CommonSshPermissionSpec & {
+  zone: string;
+  provider: "gcloud";
+  resource: {
```

Comment:
Nit: Should `provider` be the first field here? That seems to be the pattern I've observed with other discriminated types.